pygeos/test/test_constructive.py

```diff
@@ -480,6 +480,99 @@ def test_polygonize_missing():
     assert result == pygeos.Geometry("GEOMETRYCOLLECTION EMPTY")
 
 
+def test_polygonize_full():
+    lines = [
+        pygeos.Geometry("LINESTRING (0 0, 1 1)"),
+        pygeos.Geometry("LINESTRING (0 0, 0 1)"),
+        pygeos.Geometry("LINESTRING (0 1, 1 1)"),
+        pygeos.Geometry("LINESTRING (1 1, 1 0)"),
+        pygeos.Geometry("LINESTRING (1 0, 0 0)"),
+        pygeos.Geometry("LINESTRING (5 5, 6 6)"),
+        pygeos.Geometry("LINESTRING (1 1, 100 100)"),
+        pygeos.Geometry("POINT (0 0)"),
+        None,
+    ]
+    result = pygeos.polygonize_full(lines)
+    assert len(result) == 4
+    assert all(pygeos.get_type_id(geom) == 7 for geom in result)  # GeometryCollection
+    polygons, cuts, dangles, invalid = result
+    expected_polygons = pygeos.Geometry(
+        "GEOMETRYCOLLECTION (POLYGON ((0 0, 1 1, 1 0, 0 0)), POLYGON ((1 1, 0 0, 0 1, 1 1)))"
+    )
+    assert polygons == expected_polygons
+    assert cuts == pygeos.Geometry("GEOMETRYCOLLECTION EMPTY")
+    expected_dangles = pygeos.Geometry(
+        "GEOMETRYCOLLECTION (LINESTRING (1 1, 100 100), LINESTRING (5 5, 6 6))"
+    )
+    assert dangles == expected_dangles
+    assert invalid == pygeos.Geometry("GEOMETRYCOLLECTION EMPTY")
+
+
+def test_polygonize_full_array():
+    lines = [
+        pygeos.Geometry("LINESTRING (0 0, 1 1)"),
+        pygeos.Geometry("LINESTRING (0 0, 0 1)"),
+        pygeos.Geometry("LINESTRING (0 1, 1 1)"),
+    ]
+    expected = pygeos.Geometry(
+        "GEOMETRYCOLLECTION (POLYGON ((1 1, 0 0, 0 1, 1 1)))"
+    )
+    result = pygeos.polygonize_full(np.array(lines))
+    assert len(result) == 4
+    assert all(isinstance(geom, pygeos.Geometry) for geom in result)
+    assert result[0] == expected
+
```

Comment:
Here and below, might be good to assert that the last 3 parts of the result are empty geometry collections.

Reply:
Added